README.md
hello
=====

A sample CLI

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/hello.svg)](https://npmjs.org/package/hello)
[![Downloads/week](https://img.shields.io/npm/dw/hello.svg)](https://npmjs.org/package/hello)
[![License](https://img.shields.io/npm/l/hello.svg)](https://github.com/grof/hello-cli/blob/master/package.json)

<!-- toc -->
* [Usage](#usage)
* [Commands](#commands)
<!-- tocstop -->
# Usage
<!-- usage -->
```sh-session
$ npm install -g @fgroleau/grofcli
$ hello COMMAND
running command...
$ hello (-v|--version|version)
@fgroleau/grofcli/1.3.0 linux-x64 node-v16.14.0
$ hello --help [COMMAND]
USAGE
  $ hello COMMAND
...
```
<!-- usagestop -->

This is a new CLI...
A new line.
New line 2.

# Commands
<!-- commands -->
* [`hello hello [FILE]`](#hello-hello-file)
* [`hello help [COMMAND]`](#hello-help-command)

## `hello hello [FILE]`

describe the command here

```
USAGE
  $ hello hello [FILE]

OPTIONS
  -f, --force
  -h, --help       show CLI help
  -n, --name=name  name to print

EXAMPLE
  $ hello hello
  hello world from ./src/hello.ts!
```

<<<<<<< HEAD
_See code: [src/commands/hello.ts](https://github.com/grof/hello-cli/blob/v1.3.0-beta.4/src/commands/hello.ts)_
||||||| 3277176
_See code: [src/commands/hello.ts](https://github.com/grof/hello-cli/blob/v1.3.0-beta.2/src/commands/hello.ts)_
=======
_See code: [src/commands/hello.ts](https://github.com/grof/hello-cli/blob/v1.3.0/src/commands/hello.ts)_
>>>>>>> main

## `hello help [COMMAND]`

display help for hello

```
USAGE
  $ hello help [COMMAND]

ARGUMENTS
  COMMAND  command to show help for

OPTIONS
  --all  see all commands in CLI
```

_See code: [@oclif/plugin-help](https://github.com/oclif/plugin-help/blob/v3.2.18/src/commands/help.ts)_
<!-- commandsstop -->


CHANGELOG.md
# [1.3.0](https://github.com/grof/hello-cli/compare/v1.2.0...v1.3.0) (2022-03-31)


### Features

* add features A, B and C ([#30](https://github.com/grof/hello-cli/issues/30)) ([3277176](https://github.com/grof/hello-cli/commit/32771760ddc386a7619cfd283957c3747b356ebb)), closes [#27](https://github.com/grof/hello-cli/issues/27) [#28](https://github.com/grof/hello-cli/issues/28) [#29](https://github.com/grof/hello-cli/issues/29)
* introduce features A, B, C and D. ([#32](https://github.com/grof/hello-cli/issues/32)) ([d7134ac](https://github.com/grof/hello-cli/commit/d7134acb2323a71dc5c3cad7102d1eeffc9ad9e3)), closes [#27](https://github.com/grof/hello-cli/issues/27) [#28](https://github.com/grof/hello-cli/issues/28) [#29](https://github.com/grof/hello-cli/issues/29) [#31](https://github.com/grof/hello-cli/issues/31)

# [1.3.0-beta.4](https://github.com/grof/hello-cli/compare/v1.3.0-beta.3...v1.3.0-beta.4) (2022-03-31)


### Features

* add feature D from beta ([#31](https://github.com/grof/hello-cli/issues/31)) ([baf3f75](https://github.com/grof/hello-cli/commit/baf3f7596a761f9d3bac06909fd083d08e0b0f62))

# [1.3.0-beta.3](https://github.com/grof/hello-cli/compare/v1.3.0-beta.2...v1.3.0-beta.3) (2022-03-31)


### Features

* add features A, B and C ([#30](https://github.com/grof/hello-cli/issues/30)) ([3277176](https://github.com/grof/hello-cli/commit/32771760ddc386a7619cfd283957c3747b356ebb)), closes [#27](https://github.com/grof/hello-cli/issues/27) [#28](https://github.com/grof/hello-cli/issues/28) [#29](https://github.com/grof/hello-cli/issues/29)

# [1.3.0-beta.2](https://github.com/grof/hello-cli/compare/v1.3.0-beta.1...v1.3.0-beta.2) (2022-03-31)


### Features

* add feature C from beta ([#29](https://github.com/grof/hello-cli/issues/29)) ([fda0643](https://github.com/grof/hello-cli/commit/fda06436539bb0a5c7d29da207d9aa945c22ea21))

# [1.3.0-beta.1](https://github.com/grof/hello-cli/compare/v1.2.0...v1.3.0-beta.1) (2022-03-31)


### Features

* add feature A from beta branch ([#27](https://github.com/grof/hello-cli/issues/27)) ([48ef596](https://github.com/grof/hello-cli/commit/48ef59688401bee5b1a36c4d39d2c6372ccf97cd))
* add feature B from beta ([#28](https://github.com/grof/hello-cli/issues/28)) ([51586d6](https://github.com/grof/hello-cli/commit/51586d620993f003aaf67029b3355c44a0aef378))

# [1.2.0](https://github.com/grof/hello-cli/compare/v1.1.0...v1.2.0) (2022-03-31)


### Features

* add feature A ([b468e6a](https://github.com/grof/hello-cli/commit/b468e6a8aa5ce59bea177cbb85359babe3a67e51))
* add features A, B and C ([#24](https://github.com/grof/hello-cli/issues/24)) ([da5c167](https://github.com/grof/hello-cli/commit/da5c167aadd10f43c5d072a48fbe325954d2df3d)), closes [#16](https://github.com/grof/hello-cli/issues/16) [#20](https://github.com/grof/hello-cli/issues/20) [#21](https://github.com/grof/hello-cli/issues/21) [#23](https://github.com/grof/hello-cli/issues/23)
* introduce features A, B and C ([#25](https://github.com/grof/hello-cli/issues/25)) ([8283c30](https://github.com/grof/hello-cli/commit/8283c303d36fe4c53b4e82f54037bf82135f0171)), closes [#16](https://github.com/grof/hello-cli/issues/16) [#20](https://github.com/grof/hello-cli/issues/20) [#21](https://github.com/grof/hello-cli/issues/21) [#23](https://github.com/grof/hello-cli/issues/23)
* new feature C ([#22](https://github.com/grof/hello-cli/issues/22)) ([06b12da](https://github.com/grof/hello-cli/commit/06b12da707b65e364a876fd07bdf97b46f9a8cd4)), closes [#16](https://github.com/grof/hello-cli/issues/16) [#20](https://github.com/grof/hello-cli/issues/20) [#21](https://github.com/grof/hello-cli/issues/21)

# [1.2.0-beta.4](https://github.com/grof/hello-cli/compare/v1.2.0-beta.3...v1.2.0-beta.4) (2022-03-31)


### Features

* new feature C ([#22](https://github.com/grof/hello-cli/issues/22)) ([06b12da](https://github.com/grof/hello-cli/commit/06b12da707b65e364a876fd07bdf97b46f9a8cd4)), closes [#16](https://github.com/grof/hello-cli/issues/16) [#20](https://github.com/grof/hello-cli/issues/20) [#21](https://github.com/grof/hello-cli/issues/21)

# [1.2.0-beta.3](https://github.com/grof/hello-cli/compare/v1.2.0-beta.2...v1.2.0-beta.3) (2022-03-31)


### Features

* new feature C ([#23](https://github.com/grof/hello-cli/issues/23)) ([fa1eb6b](https://github.com/grof/hello-cli/commit/fa1eb6b207ce9b4ae591153d6e5fbe52bf82ff39))

# [1.2.0-beta.2](https://github.com/grof/hello-cli/compare/v1.2.0-beta.1...v1.2.0-beta.2) (2022-03-31)


### Features

* new feature B ([#21](https://github.com/grof/hello-cli/issues/21)) ([ec7a7df](https://github.com/grof/hello-cli/commit/ec7a7dfc4d4bcd911e32fc1dc88679aca42ce6bd))

# [1.2.0-beta.1](https://github.com/grof/hello-cli/compare/v1.1.0...v1.2.0-beta.1) (2022-03-31)


### Features

* a new feature ([#16](https://github.com/grof/hello-cli/issues/16)) ([aa4a9e3](https://github.com/grof/hello-cli/commit/aa4a9e3f9c6ffb32f0088dab70d2ed290726135d))
* add new feature A ([#20](https://github.com/grof/hello-cli/issues/20)) ([2544bcd](https://github.com/grof/hello-cli/commit/2544bcd821c252d5326bc605349ed18df332440d))

# [1.1.0-beta.1](https://github.com/grof/hello-cli/compare/v1.0.0...v1.1.0-beta.1) (2022-03-31)


# [1.1.0](https://github.com/grof/hello-cli/compare/v1.0.0...v1.1.0) (2022-03-31)

### Features

* a new feature ([#16](https://github.com/grof/hello-cli/issues/16)) ([aa4a9e3](https://github.com/grof/hello-cli/commit/aa4a9e3f9c6ffb32f0088dab70d2ed290726135d))


release.config.js
/* eslint-disable unicorn/prefer-module */
module.exports = {
  branches: [
    'main',
    {
      name: 'beta',
      channel: 'beta',
      prerelease: true,
    },
  ],
  // verifyConditions: [
  //   '@semantic-release/changelog',
  //   '@semantic-release/npm',
  //   '@semantic-release/git',
  // ],
  // prepare: [
  //   '@semantic-release/changelog',
  //   '@semantic-release/npm',
  //   {
  //     path: '@semantic-release/git',
  //     // assets: ['package.json', 'CHANGELOG.md', 'README.md'],
  //     assets: ['CHANGELOG.md', 'README.md'],
  //   },
  // ],
  plugins: [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    [
      '@semantic-release/changelog',
      {
        changelogFile: 'CHANGELOG.md',
      },
    ],
    '@semantic-release/npm',
    '@semantic-release/github',
    [
      '@semantic-release/git',
      {
        assets: ['CHANGELOG.md', 'README.md'],
        // assets: ['package.json'],
        // message: 'chore(release): ${nextRelease.version} [skip ci]',
        // message: 'chore(release): set `package.json` to ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}',
        // eslint-disable-next-line no-template-curly-in-string
        message: 'chore(release): set `package.json` to ${nextRelease.version} [skip ci]',
      },
    ],
  ],
}


src/commands/hello.ts
import {Command, flags} from '@oclif/command'

export default class Hello extends Command {
  static description = 'describe the command here'

  static examples = [
    `$ hello hello
hello world from ./src/hello.ts!
`,
  ]

  static flags = {
    help: flags.help({char: 'h'}),
    // flag with a value (-n, --name=VALUE)
    name: flags.string({char: 'n', description: 'name to print'}),
    // flag with no value (-f, --force)
    force: flags.boolean({char: 'f'}),
  }

  static args = [{name: 'file'}]

  async run() {
    const {args, flags} = this.parse(Hello)

    const name = flags.name ?? 'world'
    this.log(`hello ${name} from ./src/commands/hello.ts`)
    if (args.file && flags.force) {
      this.log(`you input --force and --file: ${args.file}`)
    }
  }
}

// comment 1
// comment 2
// comment 3
// comment 4
// comment f-2022-03-31-01

// add feature A on beta branch
// add feature B on beta branch
// add feature C on beta branch
// add feature D on beta branch

